systests/cluster/test_debug.py

```diff
@@ -0,0 +1,101 @@
+"""
+This module contains tests that are useful for developer debugging
+and can contain sleep statements or test that intentionally fail or break things.
+They're separate from test_remote_account.py for that reason.
+"""
+import time
+
+from ducktape.mark import matrix, parametrize
+from ducktape.mark.resource import cluster
+from ducktape.tests.test import Test
+from systests.cluster.test_remote_account import GenericService
+
+
+class FailingTest(Test):
+    """
+    The purpose of this test is to validate reporters. Some of them are intended to fail.
+    """
+    def setup(self):
+        self.service = GenericService(self.test_context, 1)
+
+    @cluster(num_nodes=1)
+    @matrix(string_param=['success-first', 'fail-second', 'fail-third'], int_param=[10, 20, -30])
+    def matrix_test(self, string_param, int_param):
+        assert not string_param.startswith('fail') and int_param > 0
+
+    @cluster(num_nodes=1)
+    @parametrize(string_param='success-first', int_param=10)
+    @parametrize(string_param='fail-second', int_param=-10)
+    def parametrized_test(self, string_param, int_param):
+        assert not string_param.startswith('fail') and int_param > 0
+
+    @cluster(num_nodes=1)
+    def failing_test(self):
+        assert False
+
+    @cluster(num_nodes=1)
+    def successful_test(self):
+        assert True
+
+
+class DebugThisTest(Test):
+
+    @cluster(num_nodes=1)
+    def one_node_test(self):
+        self.service = GenericService(self.test_context, 1)
+        self.logger.warning('one_node_test - Sleeping for 90s')
+        time.sleep(90)
+        assert True
+
+    @cluster(num_nodes=1)
+    def another_one_node_test(self):
+        self.service = GenericService(self.test_context, 1)
+        self.logger.warning('another_one_node_test - Sleeping for 30s')
+        time.sleep(30)
+        assert True
+
+    @cluster(num_nodes=1)
+    def yet_another_one_node_test(self):
+        self.service = GenericService(self.test_context, 1)
+        self.logger.warning('yet_another_one_node_test - Sleeping for 30s')
+        time.sleep(30)
+        assert True
+
+    @cluster(num_nodes=2)
+    def two_node_test(self):
+        self.service = GenericService(self.test_context, 2)
+        assert True
+
+    @cluster(num_nodes=2)
+    def another_two_node_test(self):
+        self.service = GenericService(self.test_context, 2)
+        assert True
+
+    @cluster(num_nodes=3)
+    def three_node_test(self):
+        self.service = GenericService(self.test_context, 3)
+        assert True
+
+    @cluster(num_nodes=3)
+    def another_three_node_test(self):
+        self.service = GenericService(self.test_context, 3)
+        self.logger.warning('Sleeping for 30s')
+        time.sleep(30)
+        assert True
+
+    @cluster(num_nodes=3)
+    def hanging_three_node_test(self):
```

Comment:
how does this hang?

Reply:
might be an artifact of multiple iterations